lib/Basic/Subprocess.cpp

```
@@ -436,17 +452,47 @@ void llbuild::basic::spawnProcess(
 
     // If we have been cancelled since we started, do nothing.
     if (!wasCancelled) {
-      int result = posix_spawn(&pid, args[0], /*file_actions=*/&fileActions,
-                               /*attrp=*/&attributes, const_cast<char**>(args.data()),
-                               const_cast<char* const*>(environment.getEnvp()));
+      int result = 0;
+
+#ifdef __APPLE__
+      int oldCWD = -1;
+      int chdirErr = 0;
+      if (attr.workingDir.size()) {
```

We should report an error here for non-__APPLE__, if workingDir is set.